api/proxy/config/app_config.go

```diff
@@ -47,11 +57,22 @@ func ReadAppConfig(ctx *cli.Context) (AppConfig, error) {
 		return AppConfig{}, fmt.Errorf("read proxy config: %w", err)
 	}
 
+	// NOTE: double passing of the `enabledAPIs` type to both the
+	//       the AppConfig and Rest server config is an abstraction leak.
+	//       given rest endpoint logic is guarded by the enabledAPIs it
+	//       makes sense to be like this. otherwise a separate env var
+	//       could be introduced that provides enablement toggles purely
+	//       for the REST server's code paths.
+	//
+	enabledAPIs := enabled_apis.ReadEnabledAPIs(ctx)
```

Comment:
hmm can't you embed the rest config inside the top level enabledAPIs config? what didn't work when you tried this?

Reply:
are you referring to updating the `enabledAPIs` struct to store notion of two slices (i.e, servers_enabled, rest_apis_enabled)?

Reply:
Yes. Actually just use embedde structs:
```
type RESTAPIs struct {
   admin AdminAPI
   opGeneric OPGenericAPI
   ...
}
type EnabledAPIs struct {
   RESTAPIs,
   ArbAPIs,
   m MetricsAPI
```

then EnabledAPIs contains all the apis as direct fields (via the struct embedding), but we can also ONLY pass RESTAPIs struct to REST handler, instead of the entire EnabledAPIs struct.

Reply:
8bdf3fb070ad1ccc61af03d085e493574cc81fec